execution/function/histogram.go

```diff
@@ -134,8 +134,12 @@ func (o *histogramOperator) processInputSeries(vectors []model.StepVector) ([]mo
 
 		step := o.pool.GetStepVector(vector.T)
 		for i, stepBuckets := range o.seriesBuckets {
-			// We only have one bucket so it needs to be NaN.
-			// Or, if we are after how many scalar points we have.
+			// It could be zero if multiple input series map to the same output series ID.
+			if len(stepBuckets) == 0 {
```

Comment:
Ah so this is what we miss by removing the check for`len(stepBuckets) < 2` :)

Reply:
Yeah :smile: